vllm/v1/engine/async_llm.py

```diff
@@ -52,7 +50,28 @@ def __init__(
         use_cached_outputs: bool = False,
         log_requests: bool = True,
         start_engine_loop: bool = True,
+        stat_loggers: Optional[list[type[StatLoggerBase]]] = None,
```

Comment:
@markmc I wonder whether we should change `type[StatLoggerBase]` to `Callable[[VllmConfig, int], StatLoggerBase]`, and could define a shorter type name for that like
```python
StatLoggerFactory = Callable[[VllmConfig, int], StatLoggerBase]
```

which is more general but also captures the args - you could still then pass a subclass of `StatLoggerBase` which has the appropriate constructor?

Of course still making clear that this isn't yet a stable interface.

Reply:
Nice idea; you can pull another commit from the same branch with that change :+1: